libpromises/evalfunction.c

```diff
@@ -2745,6 +2810,21 @@ static FnCallResult FnCallMapData(EvalContext *ctx, ARG_UNUSED const Policy *pol
         return FnFailure();
     }
 
+    if (mapdatamode && json_pipemode)
+    {
+        JsonElement *returnjson_pipe = ExecJSON_Pipe(arg_map, container);
+
+        if (NULL == returnjson_pipe)
+        {
+            Log(LOG_LEVEL_ERR, "Function %s failed to get output from 'json_pipe' execution", fp->name);
+            return FnFailure();
+        }
+
+        JsonDestroy(container);
```

Comment:
The `VarNameOrInlineToJson()` semantics have changed since this was made, you'll need to use the `JsonDestroyMaybe()` here.